OrbitCore/CallstackData.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CallstackData.h"
+
+#include "Callstack.h"
+#include "Capture.h"
+#include "SamplingProfiler.h"
+
+using orbit_client_protos::CallstackEvent;
+
+void CallstackData::AddCallstackEvent(CallstackEvent&& callstack_event) {
+  CallstackID hash = callstack_event.callstack_hash();
+  CHECK(HasCallStack(hash));
+  callstack_events_.push_back(std::move(callstack_event));
```

Comment:
Any reason why we need a lock below, but not here?

Reply:
We use lock only for unique_callstacks_, here we change only callstack_events_

Reply:
But should the callstack_events_ also be guarded by a lock?

Reply:
Probably yes... But we should understand if it implies on performance...